### ISSUE TITLE
Training with the CrossEntropy loss on a regression dataset fails silently

### ISSUE THREAD

**Issue by fKunstner** [fKunstner/optexp#19]
Example configuration below. It will run, but the result is not meaningful. 
I'm surprised it doesn't hit a pytorch type error - doesn't CrossEntropy expect `y` to be `long`?

Anything we can check for silly mistakes like this?

```
from optexp.cli import cli
from optexp.datasets import DummyRegression
from optexp.experiment import Experiment
from optexp.hardwareconfig.strict_manual import StrictManualConfig
from optexp.metrics.metrics import Accuracy, CrossEntropy
from optexp.models import Linear
from optexp.optim.sgd import SGD
from optexp.problem import Problem
from optexp.runner.slurm.slurm_config import SlurmConfig

experiments = [
    Experiment(
        optim=SGD(lr=lr, momentum=momentum),
        problem=Problem(
            dataset=DummyRegression(),
            model=Linear(),
            lossfunc=CrossEntropy(),
            metrics=[CrossEntropy(), Accuracy()],
            batch_size=100,
        ),
        group="testing",
        eval_every=1,
        seed=0,
        steps=10,
        hardware_config=StrictManualConfig(
            num_devices=1,
            micro_batch_size=10,
            eval_micro_batch_size=10,
            device="cpu",
        ),
    )
    for lr in [10**-2, 10**-1, 10**0]
    for momentum in [0, 0.9]
]

SLURM_CONFIG = SlurmConfig(hours=1, gb_ram=8, n_cpus=1, n_gpus=1, gpu=True)
if __name__ == "__main__":
    cli(experiments, slurm_config=SLURM_CONFIG)
```
